### PR TITLE
supermariowar: new recipe

### PR DESCRIPTION
**Pull request by Crestwave**
Notes:
- Depends on #4753 
- I was going to put this in `games-arcade`, but Gentoo had it in `games-action`
- I'm not sure if the unzipping should be done in another function
- Am I supposed to put the apps in a directory in `$appsDir`?
- I wasn't sure what to name the version; would it be preferred to make it a `~git` package?
- I create shell scripts to run them because it's designed to have the data directory path passed as an argument, but because of that it uses the icon for text files and has a different name on startup. Does this mean I have to patch the program?
- Should I enable `USE_PNG_SAVE`?
- I was also able to run the original version (this is an unofficial fork); would a recipe for it be accepted if I feel like making one in the future?

### CODE REVIEW

File: games-action/supermariowar/supermariowar-beta1.1.recipe [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+INSTALL()
+{
+	cd build
+	make install
+
+	install -m 755 -d "$appsDir" "$commandBinDir"
+	install -m 755 Binaries/Release/smw-server "$commandBinDir"
+	set smw SMW "Super Mario War" \
```
Comment (extrowerk) [inline games-action/supermariowar/supermariowar-beta1.1.recipe]: What does this do?

File: games-action/supermariowar/supermariowar-beta1.1.recipe [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+		smw-leveledit SMW-LevelEdit "Super Mario War Level Editor" \
+		smw-worldedit SMW-WorldEdit "Super Mario War World Editor"
+
+	while [ $# != 0 ]; do
```
Comment (extrowerk) [inline games-action/supermariowar/supermariowar-beta1.1.recipe]: And this one?

File: games-action/supermariowar/supermariowar-beta1.1.recipe (hunk 1 above)
Comment (Crestwave) [reply games-action/supermariowar/supermariowar-beta1.1.recipe]: The `set` command sets its arguments to positional parameters. So the first argument would be `$1`, the second `$2`, etc.

File: games-action/supermariowar/supermariowar-beta1.1.recipe (hunk 2 above)
Comment (Crestwave) [reply games-action/supermariowar/supermariowar-beta1.1.recipe]: `$#` is the number of positional parameters. Basically, every loop it uses a batch of three parameters, then it `shift`s it away to the next batch until there are none left.

**Comment by Crestwave** [comment]
I went ahead and patched the source instead of using the shell script launchers, simplifying the recipe a lot, but I still use the looping you commented on.

File: games-action/supermariowar/supermariowar-beta1.1.recipe (hunk 2 above)
Comment (extrowerk) [reply games-action/supermariowar/supermariowar-beta1.1.recipe]: Aha! Thanks for the explanation!

**Pull request merged**